Flaky test: AtomApplication launch when closing the last window quits the application

In our Windows x86 builds, we sometimes see the build fail at [this point in the test suite](https://github.visualstudio.com/Atom/_build/results?buildId=33073&view=logs&jobId=97a617bf-bcbd-5dfa-bba2-cfba2747b693&taskId=1e1369e9-1c56-59b0-830c-f1606a680c8a&lineStart=64&lineEnd=67&colStart=1&colEnd=32)...

```
  ...
  . AtomApplication launch when the `--wait` flag is passed kills the specified pid after a newly-opened directory in an existing window is closed: 8005ms
    AtomApplication launch when closing the last window quits the application:
  . AtomApplication launch when closing the last window quits the application: 7243ms
##[error]Cmd.exe exited with code '1'.
##[section]Finishing: Run tests
```

Rebuilding can sometimes resolve the failure.

---

**Affected CI provider(s)**: Azure DevOps

**Affected platform(s)**: Windows x86

**Example commit experiencing flakiness**: https://github.com/atom/atom/commit/24ed380e786fa23af761a51e5afded58fe3ca1b5

**Example failing build at that commit**: https://github.visualstudio.com/Atom/_build/results?buildId=33073

**Example passing build at that commit**: https://github.visualstudio.com/Atom/_build/results?buildId=33116

**Example log from failing build**: https://github.com/atom/atom/files/2967685/build.log

--- a/spec/main-process/atom-application.test.js
+++ b/spec/main-process/atom-application.test.js
@@ -663,6 +663,10 @@ describe('AtomApplication', function () {
             parseCommandLine([path.join(makeTempDir('a'), 'file-a')])
           )
           await focusWindow(window)
+
+          // Choosing "Don't save"
+          mockElectronShowMessageBox({ response: 2 })
+
           window.close()
           await window.closedPromise
           await atomApplication.lastBeforeQuitPromise
@@ -675,6 +679,10 @@ describe('AtomApplication', function () {
             parseCommandLine([path.join(makeTempDir('a'), 'file-a')])
           )
           await focusWindow(window)
+
+          // Choosing "Don't save"
+          mockElectronShowMessageBox({ response: 2 })
+
           window.close()
           await window.closedPromise
           await timeoutPromise(1000)

